lib/logstash/codecs/avro.rb

```diff
@@ -48,6 +51,9 @@
 class LogStash::Codecs::Avro < LogStash::Codecs::Base
   config_name "avro"
 
+  include LogStash::PluginMixins::ECSCompatibilitySupport(:disabled, :v1, :v8 => :v1)
+  include LogStash::PluginMixins::EventSupport::EventFactoryAdapter
```

Comment:
didn't add `target` option because this codec is mainly used in Kafka. `target` needs to align with the implementation of Kafka plugins in order to respect the setting. Having `codec.target` and `target` in input/ output plugins could produce unexcepted result